### ISSUE TITLE
Risk Calculation Initialisation

### ISSUE THREAD

**Issue by mike1813** [SPYDERISK/system-modeller#50]
When the Risk Calculator starts, it loads the (previously validated) system model into its cache. It then clears old risk calculation output, runs the calculation to create new risk calculation output, and optionally saves the results in the triple store.

However, this means that if a risk calculation is carried out without saving the results, the triple store may still contain the results from some previous calculation. This can cause inconsistency between the U/I client and the service, whereby the client displays new results, but subsequent requests to the service may lead to calculations based on the old results.

The first case where this inconsistency causes serious problems is discussed in #3, where a threat path analysis could be based on the wrong risk calculation results.

The proposed solution is to change the Risk Calculator so it clears risk calculation results from the triple store before starting to run a new risk calculation. This provides a better starting position for implementing the changes discussed in #3, and also in any future risk calculation post-processing feature.

**Comment by mike1813** [comment]
After further discussion, it was concluded that issue #3 can be addressed well enough without this change. Since changing the risk calculation initialisation may cause unwanted side effects (due to the way the JenaQuerierDB cache works), I think it makes sense to put work on the initialisation sequence on hold until we find another requirement for it.

Setting the wontfix label, which can be removed it the issue is 'reactivated'.

**Comment by mike1813** [comment]
Reviewed by @mike1813 : this is a bug, and although previously labelled wontfix, it probably should be addressed.

While the U/I issue has been addressed, it may still lead to incorrect threat path output being generated. This is more important than it seemed in 2023, because we now have more threat path inspection features in the U/I, and report generation is now based far more on threat path analysis than it was previously. Incorrect threat path data is likely to cause problems for reporting feature developers that are difficult to diagnose, and may lead to a lot of time being wasted.

**Comment by sdm2d11** [comment]
@sjt1970 please check if this is relevant for Telemetry / Nemecys, so we can assign it to that project.
